resharper/resharper-unity/src/Rider/UnityPluginInstaller.cs

```diff
@@ -114,12 +114,29 @@ private void InstallPluginIfRequired()
             if (versionForSolution >= new Version("2019.2")) // 2019.2+ would not work fine either without Rider package, and when package is present it loads EditorPlugin directly from Rider installation.
             {
                 var installationInfoToRemove = myDetector.GetInstallationInfo(myCurrentVersion, previousInstallationDir: FileSystemPath.Empty);
-                if (installationInfoToRemove.PluginDirectory.ExistsDirectory)
+                var pluginDll =
+                    installationInfoToRemove.PluginDirectory.Combine(PluginPathsProvider.BasicPluginDllFile);
+                if (pluginDll.ExistsFile)
                 {
                     myQueue.Enqueue(() =>
                     {
-                        myLogger.Info($"Remove {installationInfoToRemove.PluginDirectory}. Rider package should be used instead.");
-                        installationInfoToRemove.PluginDirectory.Delete();
+                        myLogger.Info($"Remove {pluginDll}. Rider package should be used instead.");
+                        pluginDll.DeleteFile();
+                        FileSystemPath.Parse(pluginDll.FullPath + ".meta").DeleteFile();
+                        
+                        // jetbrainsDir is usually "Assets\Plugins\Editor\JetBrains", however custom locations were also possible   
+                        var jetbrainsDir = installationInfoToRemove.PluginDirectory;
+                        if (jetbrainsDir.GetChildren().Any()) return;
```

Comment:
It's worth checking that the folders we're about to delete are the expected folders, so maybe check the folder name, too? Or just delete all empty folders up to `Assets`?

Reply:
Ok, I have added check for each folder name. We would not delete any folder with unexpected name.